pywbem_mock/_wbemconnection_mock.py

```diff
@@ -119,6 +110,244 @@ def _cvt_obj_name(objname):
+            if not isinstance(provider, MethodProvider):
+                raise TypeError(
+                    _format("Provider argument {0!A} is not a "
+                            "valid subclass of MethodProvider. ",
+                            provider))
+        else:
+            raise ValueError(
+                _format("Provider_type argument {0!A} is not a valid provider "
+                        "type. Valid provider types are {1!A}.",
+                        provider_type, provider_types))
+
+        if provider_classnames is None:
+            raise ValueError(
+                _format('Classnames argument must be string '
+                        'or list of strings. None not allowed.'))
+
+        if namespaces is None:
+            namespaces = [conn.default_namespace]
+
+        if isinstance(namespaces, six.string_types):
+            namespaces = [namespaces]
+
+        if not isinstance(namespaces, (list, tuple)):
+            raise TypeError(
+                _format('Namespace argument {0|A} must be a string or '
+                        'list/tuple but is {1}',
+                        namespaces, type(namespaces)))
+
+        for namespace in namespaces:
+            if not isinstance(namespace, six.string_types):
+                raise TypeError(
+                    _format('Namespace "{0!A}" in namespaces argument not '
+                            'a string. ', namespace))
+
+            if namespace not in conn.namespaces:
+                raise ValueError(
+                    _format('Namespace "{0!A}" in namespaces argument not '
+                            'in CIM repository. '
+                            'Existing namespaces are: {1!A}. ',
+                            namespace, conn.namespaces))
+
+        if isinstance(provider_classnames, six.string_types):
+            provider_classnames = [provider_classnames]
+        assert isinstance(provider_classnames, (list, tuple))
+        for classname in provider_classnames:
+            assert isinstance(classname, six.string_types)
+
+            # For each namespace in which the provider is to be registered,
+            # if the class is not in that namespace, either compile it in if
+            # pragmas exist or generate an exception if no pragmas exist
+
+            for namespace in namespaces:
+                if not conn.mainprovider.class_exists(namespace, classname):
+                    if schema_pragma_files:
+                        conn.compile_schema_classes(
+                            classname,
+                            schema_pragma_files,
+                            namespace=namespace,
+                            verbose=False)
+                    else:
+                        raise ValueError(
+                            _format('Class "{0!A}" does not exist in '
+                                    'namespace {1!A} of the CIM repository '
+                                    'and no pragma files',
```

Comment:
The last part of this message can be misunderstood to mean the class is not in any schema pragma files, but it really means there were no schema pragma files specified.
Suggest to change it to: "and no schema pragma files were specified"

Reply:
DONE